src/test/java/com/flab/bbt/aop/LoginCheckAspectUnitTest.java

```diff
@@ -0,0 +1,86 @@
+package com.flab.bbt.aop;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import com.flab.bbt.AbstractContainerBaseTest;
+import com.flab.bbt.ConfigureTestProfile;
+import com.flab.bbt.common.SessionConst;
+import com.flab.bbt.exception.CustomException;
+import com.flab.bbt.exception.ErrorCode;
+import com.flab.bbt.payment.controller.PaymentController;
+import com.flab.bbt.payment.request.PaymentRequest;
+import com.flab.bbt.payment.service.PaymentService;
+import com.flab.bbt.user.domain.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.aop.aspectj.annotation.AspectJProxyFactory;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.mock.web.MockHttpSession;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@ConfigureTestProfile
+@ExtendWith(MockitoExtension.class)
+public class LoginCheckAspectUnitTest extends AbstractContainerBaseTest {
+
+    private PaymentController paymentController;
+
+    @MockBean
+    private PaymentService paymentService;
+
+    MockHttpSession mockHttpSession;
+
+    @BeforeEach
+    void beforeEach() {
+        mockHttpSession = new MockHttpSession();
+        final PaymentController target = new PaymentController(paymentService);
+
+        final AspectJProxyFactory proxyFactory = new AspectJProxyFactory(target);
+        proxyFactory.addAspect(new LoginCheckAspect(mockHttpSession));
+
+        paymentController = proxyFactory.getProxy();
+    }
```

Comment:
유닛테스트로 잘 구현해주셨네요 👍 

조금만 더 보충하자면..
1. `PaymentController`를 모킹하자. -> aop에서 호출하는 대상들만 모킹하면됩니다.
2. `paymentController` 네이밍변경 : `proxyPaymentController` -> 


```suggestion
    @BeforeEach
    void beforeEach() {
        mockHttpSession = new MockHttpSession();
        final PaymentController target = mock(PaymentController.class);

        final AspectJProxyFactory proxyFactory = new AspectJProxyFactory(target);
        proxyFactory.addAspect(new LoginCheckAspect(mockHttpSession));

        proxyPaymentController = proxyFactory.getProxy();
    }
```

Reply:
@f-lab-lion 
변수명 변경해두었습니다.

1번에서 제안해주신 모킹하는 방법은 ClassCastException 이 발생합니다. 이유를 생각해봤는데,
`mock(PaymentController.class)`을 하게 되면 mock용도의 프록시 객체가 생성이 됩니다. 이 객체를 또 한 번 aop용도의 프록시 객체로 생성하는 과정(`new AspectJProxyFactory(target);`)에서 `java.lang.reflect.Proxy`를 부모로 하는 프록시 객체가 생성되어 PaymentController로의 캐스팅이 안되는 것으로 보입니다. 

이러한 이유로 해당 부분은 모킹없이 진행하였습니다.